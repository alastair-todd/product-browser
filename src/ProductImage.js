import React from 'react'

class ProductImage extends React.Component {
  render () {
    return this.props.imageUrl
      ? <img src={this.props.imageUrl} />
      : <div className="no-image" />
  }
}
ProductImage.propTypes = {
  imageUrl: React.PropTypes.string
}
export default ProductImage
