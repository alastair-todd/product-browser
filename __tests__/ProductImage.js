// ProductImage.react-test.js
import React from 'react'
import ProductImage from '../src/ProductImage'
import renderer from 'react-test-renderer'

jest.mock('react-dom')

describe('ProductImage', () => {
  it('is rendered', () => {
    const component = renderer.create(
      <ProductImage imageUrl="assets/img/bear2.png" />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})