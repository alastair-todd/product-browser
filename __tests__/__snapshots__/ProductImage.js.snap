exports[`ProductImage is rendered 1`] = `
<img
  className="img-responsive"
  src="assets/img/bear2.png" />
`;
